### PR TITLE
Creating Custom Extendable and Expandable Floating Action Button In Android Using Kotlin

### LINKED ISSUE #3974: [Android] Creating Custom Extendable and Expandable Floating Action Button In Android Using Kotlin
#### Proposed title of article
 Creating Custom Extendable and Expandable Floating Action Button In Android Using Kotlin

#### Proposed article introduction
A floating action button (FAB) is a circular button that triggers the primary action in your app's UI. A FAB performs the primary, or most common, action on a screen. It appears in front of all screen content, typically as a circular shape with an icon in its center.

#### Key takeaways 
- What is FAB
- How to create extendable FAB
- How to create expandable FAB

#### Article quality
My article is going to be unique since am going to use material deign and as well implement it using Kotlin programming language

#### References
- https://material.io/components/buttons-floating-action-button/android
- https://developer.android.com/guide/topics/ui/floating-action-button

### PR DESCRIPTION
**Pull request by omondi-alex**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #3974

### CODE REVIEW

**Comment by Kayere** [comment]
Plagiarism check done ✔️

File: content/articles/creating-custom-expandable-fab/index.md [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+        android:fromYDelta="300%"
+        android:toYDelta="0%" />
+
+    <scale
+        android:pivotY="50%"
+        android:pivotX="50%"
+        android:toXScale="0.9"
+        android:toYScale="0.9"/>
+
+    <alpha
```
Comment (Ericgacoki) [inline content/articles/creating-custom-expandable-fab/index.md:108]: Explain what these tags, attributes and values mean.

File: content/articles/creating-custom-expandable-fab/index.md [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
+import android.view.animation.Animation
+import android.view.animation.AnimationUtils
+import android.widget.Toast
+import com.example.expandablefabdemo.databinding.ActivityMainBinding
```
Comment (Ericgacoki) [inline content/articles/creating-custom-expandable-fab/index.md:195]: I don't think it is necessary to include the package name and imports. Android Studio can handle that.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Hello @omondi-alex, check out the following comments and ping me when you're done!

**Comment by omondi-alex** [comment]
Done with the corrections

**Review by Ericgacoki — COMMENTED** [review]
Hello @omondi-alex, this is quite impressive and easy to follow tutorial. However, some parts of the code aren't explained clearly and beginners might find it difficult to understand some snippets. Take a look at the comments below!

**Comment by omondi-alex** [comment]
Changes completed @Ericgacoki

**Review by Ericgacoki — APPROVED** [review]

**Comment by lalith1403** [comment]
Jumping on this one.

**Review by lalith1403 — APPROVED** [review]

**Review by lalith1403 — APPROVED** [review]

**Review by lalith1403 — APPROVED** [review]
A straightforward and easy to read tutorial. Well done :)

File: content/articles/creating-custom-expandable-fab/index.md [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+url: /creating-a-custom-fab-in-android/
+title: How to create Custom Extendable and Expandable FAB in Android Using Kotlin
+description: This tutorial will guide the reader through the process of creating a custom extendable and expandable FAB in Android using kotlin. A FAB is a circular button that usually floats on the screen.
+author: omondi-alex
+date: 2021-11-26T00:00:00-00:29
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/creating-a-custom-fab-in-android/hero.png
```
[the remark this answers is not in the document]
Comment (Ericgacoki) [reply content/articles/creating-custom-expandable-fab/index.md]: Noted :heavy_check_mark:

**Comment by omondi-alex** [comment]
> A straightforward and easy to read tutorial. Well done :)

Thanks for your reply

**Pull request merged**